pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTFormalParameters.java

```diff
@@ -24,4 +27,31 @@ public int getParameterCount() {
     public Object jjtAccept(JavaParserVisitor visitor, Object data) {
         return visitor.visit(this, data);
     }
+
+
+    @Override
+    public Iterator<ASTFormalParameter> iterator() {
```

Comment:
Interesting idea! We probably could use this for different node types, too.

Reply:
ASTTypeArguments is a good candidate. I've already implemented that in one of my branches, but the topic of the branch is not a priority for now